src/control/server/ctl_network_rpc.go

```diff
@@ -49,7 +48,16 @@ func (c *ControlService) NetworkScan(ctx context.Context, req *ctlpb.NetworkScan
 		excludeMap[iface] = struct{}{}
 	}
 
-	results, err := netdetect.ScanFabric(req.GetProvider())
+	provider := c.srvCfg.Fabric.Provider
+	switch {
+	case strings.EqualFold(req.GetProvider(), "all"):
+		provider = ""
+	case req.GetProvider() != "":
+		provider = req.GetProvider()
+	default:
```

Comment:
nit: no need for empty default case

Reply:
I just missed your comment while merging with master and pushing again.  I see you approved, but if you want me to make the change, it's no big deal and I can do that.

Reply:
I made the change in my latest.